lib/after_commit_action.rb

```diff
@@ -17,6 +17,11 @@ module AfterCommitAction
   end
 
   def execute_after_commit(&block)
+    if ActiveRecord::Base.connection.open_transactions == 0 ||
+      ActiveRecord::Base.connection.add_transaction_record(self).nil?
+      return block.call
+    end
```

Comment:
I pulled this down to poke at it and find it more readable written like this:
```
    if ActiveRecord::Base.connection.open_transactions == 0
      return block.call
    else
      ActiveRecord::Base.connection.add_transaction_record(self)
    end
```

However, I'm still concerned that we are now *always* calling `add_transaction_record` for every call. I couldn't immediately find a way to check whether an object is already registered in the transaction, or find information that says it doesn't matter to add the object multiple times.

Anything else you can add here regarding this?